### DISCUSSION TITLE
Lifelines missing data and FHIR representation of Conditions' negative cases

### DISCUSSION THREAD

**Discussion by hcadavid** [MyDigiTwinNL/MyDigiTwin-architecture-documentation#5] [General]
Hello @baukearends and @KasiaSmietanka,

This is a follow-up discussion on pairing-rules issue #22 (https://github.com/MyDigiTwinNL/CDF2Medmij-Mapping-tool/issues/22). This is the last 'issue' to be solved concerning data harmonisation (for the PoC), so I'd appreciate your input on this.

Following @baukearends suggestion, I'm defining an alternative pairing rules that allows the inclusion of Conditions (Stroke, Diabetes, etc) self-reported as not present  - so that we can distinguish them from the Conditions that are not included on the target FHIR dataset due to missing data (e.g., skipped assessments) on lifelines. However, this raised new questions, in particular on which cases we can say that a participant didn't have a given condition across the study (i.e., the condition can be considered as negative), and in which cases the condition can't be considered as neither active nor inactive, due to missing data. 

My interpretation is the following:

**IF a YES/TRUE is given on the baseline, or on any follow-up assessment, e.g.:**
``` 

                                [1A][1B][1C][2A][3A][3B] 
 stroke_presence_adu_q_1        [1 ][  ][  ][  ][  ][  ]
 stroke_followup_adu_q_1        [  ][X ][X ][X ][X ][X ]

 OR

                                [1A][1B][1C][2A][3A][3B]
 stroke_presence_adu_q_1        [X ][  ][  ][  ][  ][  ]
 stroke_followup_adu_q_1        [  ][X ][1 ][X ][X ][X ] 
 

 
``` 
**The output is:** Status: Active. Verification: Unknown (this is how it is currently mapped)
          

**ELSE IF there is an explicit NO/FALSE on the last follow up (in this example, 3b), e.g.:**
```  
                                [1A][1B][1C][2A][3A][3B]
 stroke_presence_adu_q_1        [X ][  ][  ][  ][  ][  ]
 stroke_followup_adu_q_1        [  ][X ][X ][X ][X ][2 ] Inactive/Known absent
```
**The output is:** Status: Inactive. Confirmation: Known to be absent. 
 
**ELSE (Any other: e.g.:)**
``` 
                                [1A][1B][1C][2A][3A][3B]
 stroke_presence_adu_q_1        [2 ][  ][  ][  ][  ][  ]
 stroke_followup_adu_q_1        [  ][2 ][2 ][  ][2 ][  ]
``` 
**The resource is not created (can't be identified as neither positive nor negative case).**

As you see, my rationale for the last two conditions is: when a participant has not self-reported a Condition as 'Active' (baseline or followup assessments), on the FHIR dataset this condition would be identified as 'Inactive/Absent' if and only if the last follow-up assessment was included. Any other interpretation would require making assumptions about the outcome of the missing last assessment (the last assessment with respect to the overall cohort study). 

Using this, for example, the following subsets for 'stroke' cases could be identified (using SQL) from within the harmonised data:

Self-reported as positive: 6217
Self-reported as negative:9013
Undetermined: 135883

Does this interpretation make sense to you?

**Comment by KasiaSmietanka** [comment]
Quick questions:
1. Am I correct in understanding that by 'Confirmed positive,' you are referring to individuals with a confirmed condition, such as a stroke? So, does '6217' represent the number of individuals with a confirmed stroke, for example?

2. Does 'Undetermined' refer to the number of individuals for whom a condition, such as a stroke, cannot be confirmed as either positive or negative?

Given that this data is self-reported and has not been further validated against other questionnaires or medical records, some level of inconsistency is expected. Therefore, I would be cautious about applying overly strict criteria to confirm or rule out whether a specific event occurred. However, it is interesting to observe that the self-reported Lifelines data appears to lack full reliability.

**Reply by hcadavid** [reply]
I just updated the wording in the original question, as my use of the 'confirmed' adjective was misleading in this context -as you said, confirmation involves cross-validation with other sources. As you'll see in the updated question, I'm referring to Conditions that on the FHIR dataset would be not confirmed but self-reported as positive or negative. That is to say, the 6217 represent the number of participants on the FHIR dataset that have a self-reported stroke condition, and 9013 participants that until the end of the study consistently self-reported never having the condition. 

My point of this discussion is that I believe the Stroke condition on the remaining ones can't be identified to be neither positive nor negative as in these the the last assessment of stroke_followup_adu_q_1 (3b) is missing. 

However I'm not sure if my interpretation is correct, and to what extent distinguishing the conditions that are were consistently self-reported as negative by the participants, and the ones that are not included due to the lack of details are useful for analysis or model training purposes.

**Comment by baukearends** [comment]
This issue seems mostly related to dropout during the subsequent survey rounds, and not so much related to missing values within a questionnaire, see the screenshot below. I think how we handle this issue is related to the outcomes we would like to predict. In earlier discussions we settled on (an approximation of) the 10-year risk of having a certain disease, right?

Looking at the data availability for example for stroke, one can see that between 3A and 3B two thirds of the data is lost due to dropout. However, the interval between 1A and 3A is still approximately 10 years. 

1A 2007-2013
3A 2019-2023
3B 2021-2024

![image](https://github.com/user-attachments/assets/0aa08398-15cb-4a7d-ad81-32f79d32447f)

Would it therefore be an idea to use 3A as the last observation? This hopefully leads to more cases compared to using 3B as final endpoint.

**Reply by KasiaSmietanka** [reply]
I’m wondering whether validating the self-reported outcomes is a necessary step. Patients may either exaggerate CVD-related outcomes or underreport the severity of their condition. My approach would be to extract events from a 10-year timeframe and compare them with CBS data. If discrepancies are found, we can implement additional validation for the self-reported data if necessary.

@baukearends - do you know if there is any progress on CBS linkage of the data that we reported?

**Reply by baukearends** [reply]
That's also true, just compare the findings from CBS with the responses in Lifelines (yes/no/missing). 

Regarding the CBS linkage, in my email on September 24th I asked whether you would like to check my code for any potential errors, just to make sure that we only have to do the linkage once. Since then, I have been on a holiday and got back yesterday. We can proceed and send the data to CBS today if that's what you prefer, let's dicuss the matter later this morning.
